x/evm/types/tx_args.go

```diff
@@ -80,6 +83,23 @@ func (args *TransactionArgs) ToTransaction() *MsgEthereumTx {
 
 	var data types.TxData
 	switch {
+	case args.AuthorizationList != nil:
+		al := types.AccessList{}
+		if args.AccessList != nil {
+			al = *args.AccessList
+		}
+		data = &types.SetCodeTx{
+			To:         *args.To,
+			ChainID:    uint256.MustFromBig(args.ChainID.ToInt()),
+			Nonce:      uint64(*args.Nonce),
+			Gas:        uint64(*args.Gas),
+			GasFeeCap:  uint256.MustFromBig((*big.Int)(args.MaxFeePerGas)),
```

Comment:
can't you just change the type of `SetCodeTx` to be Big instead of *uint256.Int to avoid this unsafe conversion?

Reply:
its defined in geth, existing tx types like DynamicFeeTx keeps big but new type use uint256